pkg/credentialprovider/plugin/config_test.go

```diff
@@ -376,25 +385,174 @@ providers:
+					},
+					{
+						Name:                 "test2",
+						MatchImages:          []string{"registry.io/two"},
+						DefaultCacheDuration: &metav1.Duration{Duration: 5 * time.Minute},
+						APIVersion:           "credentialprovider.kubelet.k8s.io/v1",
+					},
+				},
+			},
+		},
+		{
+			name: "directory with mixed API versions in config files",
+			configData: []string{
+				`---
+kind: CredentialProviderConfig
+apiVersion: kubelet.config.k8s.io/v1beta1
+providers:
+  - name: test-beta
+    matchImages:
+    - "beta.registry.io/*"
+    defaultCacheDuration: 15m
+    apiVersion: credentialprovider.kubelet.k8s.io/v1beta1`,
+				`---
+kind: CredentialProviderConfig
+apiVersion: kubelet.config.k8s.io/v1
+providers:
+  - name: test-v1
+    matchImages:
+    - "v1.registry.io/*"
+    defaultCacheDuration: 20m
+    apiVersion: credentialprovider.kubelet.k8s.io/v1`,
+			},
+			isDir: true,
+			config: &kubeletconfig.CredentialProviderConfig{
+				Providers: []kubeletconfig.CredentialProvider{
+					{
+						Name:                 "test-beta",
+						MatchImages:          []string{"beta.registry.io/*"},
+						DefaultCacheDuration: &metav1.Duration{Duration: 15 * time.Minute},
+						APIVersion:           "credentialprovider.kubelet.k8s.io/v1beta1",
+					},
+					{
+						Name:                 "test-v1",
+						MatchImages:          []string{"v1.registry.io/*"},
+						DefaultCacheDuration: &metav1.Duration{Duration: 20 * time.Minute},
+						APIVersion:           "credentialprovider.kubelet.k8s.io/v1",
+					},
+				},
+			},
+		},
+		{
+			name: "directory with one invalid config file",
+			configData: []string{
+				`---
+kind: CredentialProviderConfig
+apiVersion: kubelet.config.k8s.io/v1
+providers:
+  - name: test1
+    matchImages:
+    - "registry.io/one"
+    defaultCacheDuration: 10m
+    apiVersion: credentialprovider.kubelet.k8s.io/v1`,
+				`---
+kind: WrongKind
+apiVersion: kubelet.config.k8s.io/v1
+providers:
+  - name: test2
+    matchImages:
+    - "registry.io/two"
+    defaultCacheDuration: 5m
+    apiVersion: credentialprovider.kubelet.k8s.io/v1`,
+			},
+			isDir:     true,
+			config:    nil,
+			expectErr: `no kind "WrongKind" is registered for version "kubelet.config.k8s.io/v1"`,
+		},
 	}
 
 	for _, testcase := range testcases {
 		t.Run(testcase.name, func(t *testing.T) {
-			file, err := os.CreateTemp("", "config.yaml")
-			if err != nil {
-				t.Fatal(err)
-			}
-			defer utiltesting.CloseAndRemove(t, file)
+			var configPath string
+			var err error
+
+			if testcase.isDir {
+				// Create a temporary directory for multiple config files
+				tempDir, err := os.MkdirTemp("", "config-dir")
+				if err != nil {
+					t.Fatal(err)
+				}
+				defer func() {
+					if err := os.RemoveAll(tempDir); err != nil {
+						t.Errorf("failed to remove temp directory: %v", err)
+					}
+				}()
+
+				for i, configData := range testcase.configData {
+					fileName := fmt.Sprintf("config-%03d.json", i)
```

Comment:
Why is the json file extension must for the multiple config scenario?

Reply:
either extension will work as you pointed out. fixing this to yaml as the test cases are yaml